packages/matter.js/src/cluster/ColorControlCluster.ts

```diff
@@ -0,0 +1,734 @@
+    moveHue: OptionalCommand(0x01, MoveHueCommandRequest, 0x01, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.6 */
+    stepHue: OptionalCommand(0x02, StepHueCommandRequest, 0x02, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.7 */
+    moveToSaturation: OptionalCommand(0x03, MoveToSaturationCommandRequest, 0x03, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.8 */
+    moveSaturation: OptionalCommand(0x04, MoveSaturationCommandRequest, 0x04, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.9 */
+    stepSaturation: OptionalCommand(0x05, StepSaturationCommandRequest, 0x05, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.10 */
+    moveToHueAndSaturation: OptionalCommand(0x06, MoveToHueAndSaturatioCommandRequest, 0x06, TlvNoResponse),
+}
+
+const colorTemperatureCommands = {
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.14 */
+    moveToColorTemperature: OptionalCommand(0x0a, MoveToColorTemperatureCommandRequest, 0x0a, TlvNoResponse), // This and 4b, 4c Mandatory if ColorTemperature set in feature flags
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.21 */
+    moveColorTemperature: OptionalCommand(0x4b, MoveColorTemperatureRequest, 0x4b, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.22 */
+    stepColorTemperature: OptionalCommand(0x4c, StepColorTemperatureRequest, 0x4c, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.20 */
+    stopMoveStep: stopMoveStepCommand,
+}
+
+const xYCommands = {
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.11 */
+    moveToColor: OptionalCommand(0x07, MoveToColorCommandRequest, 0x07, TlvNoResponse), // This and 8,9 Mandatory if XY set in feature flags
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.12 */
+    moveColor: OptionalCommand(0x08, MoveColorCommandRequest, 0x08, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.13 */
+    stepColor: OptionalCommand(0x09, StepColorCommandRequest, 0x09, TlvNoResponse),
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 3.2.11.20 */
+    stopMoveStep: stopMoveStepCommand,
+}
+
+export const ColorControlClusterWithHueSaturation = ClusterExtend(
+    ColorControlCluster,
+    {
+        supportedFeatures: {
+            hueSaturation: true,
+            enhancedHue: false,
+            colorLoop: false,
+            xY: false,
+            colorTemperature: false,
+        },
+
+        attributes: hueSaturationAttributes,
+        commands: hueSaturationCommands
+    });
+
+export const ColorControlClusterWithHueSaturationAndEnhancedHue = ClusterExtend(
```

Comment:
Question is if, also because of specs Enhanced hue always includes hue/sat if we can not simply name to 

ColorControlEnhancedHue ( I know naming not 100% correct for features, but semantically still :-) Same for other names